DataDog/datadog-operator
# [profiles] Add profile label key to node affinity

### What does this PR do?

Add the profile label key as a node selector requirement to the node affinity of a user-profile daemonset

### Motivation

If nodes aren't labeled properly with the profile label key but a profile daemonset is created, there can be 2 node agents on one node. If this happens, only the default daemonset agent pod will be present on a single node rather than both default daemonset agent pod and profile daemonset agent pod on a single node

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: n/a
* Cluster Agent: n/a

### Describe your test plan

* Deploy the operator with profiles enabled and create a profile
* Check that the profile daemonset has as part of the node affinity:
```
              - key: agent.datadoghq.com/datadogagentprofile
                operator: In
                values:
                - <profile-name>
```

### Checklist
